tests/foreman/ui/test_myaccount.py

```diff
@@ -199,22 +176,22 @@ def test_positive_update_password(self):
         """
         for password in _valid_string_data(max_len=254):
             with self.subTest(password):
-                with self.logged_test_user():
-                    self.my_account.navigate_to_entity()
-                    self.my_account.wait_until_element(
-                        locators['users.password'])
-                    self.my_account.assign_value(
-                        locators['users.password'], password)
-                    self.my_account.assign_value(
-                        locators['users.password_confirmation'], password)
-                    self.my_account.click(common_locators['submit'])
-                    self.my_account.wait_until_element_is_not_visible(
-                        common_locators["notif.success"])
-                    self.login.logout()
-                    self.login.login(self.account_user.login, password)
+                user = User(password='old_password').create()
+                with Session(self.browser, user.login, 'old_password'):
+                    self.my_account.update(
+                        password=password, password_confirmation=password)
+
+                # UINoSuchElementError is raised on __exit__ once logout is
+                # only possible with prior login
+                with self.assertRaises(UINoSuchElementError):
+                    with Session(self.browser, user.login, 'old_password'):
+                        self.assertFalse(self.login.is_logged())
+
+                with Session(self.browser, user.login, password):
                     self.assertTrue(self.login.is_logged())
-                    # Updating test user password for next login
-                    self.account_user.password = password
+                    # Check user can navigate to her own account again
+                    self.my_account.navigate_to_entity()
+                user.delete()
```

Comment:
We do not include teardown logic into tests themselves as it can affect result itself. In case `user` entity has issues - `my_account` test will fail and that is not right approach. And basically, from test automation perspective it is not right to put cleanup actions in tests. We have full database erase between tiers and that should suffice our needs.